src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/HandlerNotFoundException.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClientX.Handlers
+{
+    /// <summary>
+    /// Exception when a handler is not defined
+    /// </summary>
+    internal class HandlerNotFoundException : Exception
```

Comment:
I will suggest if we are creating our own exception than have a two base class may PermenantException or TransitentException and let us custom exception inherit from one. This also helps differentiate our own exception vs catch system level exception.

Reply:
WHy not just throw new Excpetion ("HandlerNotFoundException")

Reply:
Transient exceptions are handled by their respective handlers (maybe there is retry involved). Permanent exceptions are immediately thrown. 
The reason for creating HandlerNotFoundException is to have a new constructor which takes the inner exceptions and implements all 4 constructors https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2015/code-quality/ca1032-implement-standard-exception-constructors?view=vs-2015&redirectedfrom=MSDN